src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckWorkerActor.scala

```diff
@@ -163,6 +154,32 @@ class HealthCheckWorkerActor(implicit mat: Materializer) extends Actor with Stri
       }
     }
   }
+
+  def singleRequest(httpRequest: HttpRequest)(implicit mat: Materializer): Future[HttpResponse] = {
+    if (httpRequest.uri.scheme.equalsIgnoreCase("https")) {
+      // This is only a health check, so we are going to allow _very_ bad SSL configuration.
+      val disabledSslConfig = AkkaSSLConfig().mapSettings(s => s.withLoose {
+        s.loose.withAcceptAnyCertificate(true)
+          .withAllowLegacyHelloMessages(Some(true))
+          .withAllowUnsafeRenegotiation(Some(true))
+          .withAllowWeakCiphers(true)
+          .withAllowWeakProtocols(true)
+          .withDisableHostnameVerification(true)
+          .withDisableSNI(true)
+      })
+      val authority = httpRequest.uri.authority
+      val connectionFlow = Http().outgoingConnectionHttps(
```

Comment:
Do we have to create the flow each time? How about creating it once?

Reply:
Actually, `Http()` is an akka extention, and it is only created once during the first use, and it's an actor system wide one. Once it's initialized, then it's instance will always be returned when we call `Http()`, so there is no need to put it in a separate val. The connection requires address and port, and since we're connecting to absolutely different nodes, we need to create it every time and can't reuse it unfortunately (BTW this actor is also short-lived, and it will die after the health check is performed).